lxd/containers.go

```diff
@@ -1719,6 +1737,16 @@ func (c *lxdContainer) setupMacAddresses(d *Daemon) error {
 			}
 		}
 
+		/*
+		 * propagate the name of the device down for nics if the user
+		 * didn't specify one. This is probably what they wanted
+		 * anyway, and lets us match against network names once they go
+		 * into lxc, e.g. to pull out the veth name for `lxc info`.
+		 */
+		if _, ok := d["name"]; !ok {
+			d["name"] = name
+		}
+
```

Comment:
Oh, then it's almost certainly wrong to do so.

"name" in the devices dict for a "nic" type device, is the name of the device inside the container.

So with your change, if I called my "nic" entry "lxcbr0", then after saving+reloading the container config (say with show | edit), the device inside the container will change from "eth0" to "lxcbr0", likely breaking a few things in the process :)